### PR TITLE
Selector Unit Tests

### PR DESCRIPTION
**Pull request by tomjackking**
- added unit tests for selector
- added an extra unit test for the form option component

### CODE REVIEW

File: components/selector/tsconfig.json [hunk 1]
```diff
@@ -15,7 +15,7 @@
 		"isolatedModules": true,
 		"noEmit": true,
 		"jsx": "react-jsx",
-		"types": ["@emotion/core"]
-	},
+		"types": ["@emotion/core", "jest", "@testing-library/jest-dom"]
+		},
```
Comment (benweier) [inline components/selector/tsconfig.json:19]:
```suggestion
	},
```

File: components/selector/tests/unit/selectorOption.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+					option{' '}
+				</SimpleOption>
+				<SimpleOption type="checkbox" hint="test checkbox hint">
+					{' '}
+					option{' '}
+				</SimpleOption>
+				<SimpleOption href="test" hint="test link hint">
+					{' '}
+					option{' '}
+				</SimpleOption>
```
Comment (benweier) [inline components/selector/tests/unit/selectorOption.spec.tsx:115]:
```suggestion
				<SimpleOption type="radio" hint="test radio hint">
					option
				</SimpleOption>
				<SimpleOption type="checkbox" hint="test checkbox hint">
					option
				</SimpleOption>
				<SimpleOption href="test" hint="test link hint">
					option
				</SimpleOption>
```

File: components/selector/tests/unit/selectorOption.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+		expect(getByText(/test secondary label/)).toBeInTheDocument();
+	});
+
+	test('should change type to link if href prop passed', () => {
+		const { getByRole } = render(<SimpleOption href="test">Test option</SimpleOption>);
+
+		expect(getByRole('link', { current: false })).toBeInTheDocument();
+	});
+
+	test('should be disabled if disable prop is passed', () => {
```
Comment (benweier) [inline components/selector/tests/unit/selectorOption.spec.tsx:95]:
```suggestion
	test('should be disabled if disabled prop is passed', () => {
```

File: components/selector/tests/unit/selector.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+
+	test('should options should be disabled if disabled prop is passed', () => {
+		const { getByTestId } = render(<SimpleSelector disabled type="checkbox" />);
+
+		expect(getByTestId('opt1')).toBeDisabled();
+		expect(getByTestId('opt2')).toBeDisabled();
+		expect(getByTestId('opt3')).toBeDisabled();
+	});
+
+	test('should work as intened if type is button', () => {
```
Comment (benweier) [inline components/selector/tests/unit/selector.spec.tsx:134]:
```suggestion
	test('should call onChange function if type is button', () => {
```

File: components/selector/tests/unit/selector.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+
+const SimpleSelectorNoOptions = (props: SelectorProps) => (
+	<GEL brand={wbc}>
+		<Selector {...props}>{props.children}</Selector>
+	</GEL>
+);
+
+// Component specific tests
+describe('Selector component', () => {
+	test('It should render Simple Selector as radio', () => {
```
Comment (benweier) [inline components/selector/tests/unit/selector.spec.tsx:91]:
```suggestion
	test('should render Simple Selector as radio', () => {
```

File: components/form-check/tests/unit/option.spec.tsx [hunk 6]
```diff
@@ -36,4 +36,10 @@ describe('Option component', () => {
 		const { getByText } = render(<SimpleOption hint="this is hint text" />);
 		expect(getByText(/this is hint text/)).toBeInTheDocument();
 	});
+
+	test('should be disabled if disable prop is passed', () => {
```
Comment (benweier) [inline components/form-check/tests/unit/option.spec.tsx:40]:
```suggestion
	test('should be disabled if disabled prop is passed', () => {
```

**Review by benweier — CHANGES_REQUESTED** [review]

File: components/selector/tsconfig.json (hunk 1 above)
Comment (tomjackking) [reply components/selector/tsconfig.json]: Fixed this before I saw the comment

**Review by benweier — APPROVED** [review]

**Review by soobin93 — APPROVED** [review]

**Pull request merged**